backend/internal/models/admin.go

```diff
@@ -0,0 +1,28 @@
+package models
+
+type Admin struct {
+	ID             string `json:"id"` // the ID of the corresponding Firebase Auth user, not a UUID
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	CanManageUsers bool   `json:"can_manage_users"`
+	CanUploadBooks bool   `json:"can_upload_books"`
+	CanDeleteBooks bool   `json:"can_delete_books"`
+	IsPrimaryAdmin bool   `json:"is_primary_admin"`
+}
+
+type CreateAdmin struct {
+	Name           string `json:"name"`
+	Email          string `json:"email"`
+	Password       string `json:"password"`
+	CanManageUsers bool   `json:"can_manage_users"`
+	CanUploadBooks bool   `json:"can_upload_books"`
+	CanDeleteBooks bool   `json:"can_delete_books"`
+	IsPrimaryAdmin bool   `json:"is_primary_admin"`
```

Comment:
This field is used for Update and Delete, to make sure that we cannot delete/update the primary admin. I could add this to the middleware, though.

Currently it is impossible to create more primary admins through the API, the database call will ignore that field and fill in false

Reply:
> Currently it is impossible to create more primary admins through the API, the database call will ignore that field and fill in false

Ah good, thanks.

> This field is used for Update and Delete, to make sure that we cannot delete/update the primary admin

Meaning you trust the user input to determine if the admin is primary? I think it would be far safer to double-check ourselves.

Reply:
Currently, Update and Delete take in an id to delete (admins/{id}). We then run getAdminByID, to check to see if the admin to update/delete even exists. If the admin does exist, then we check the isPrimaryAdmin field of the admin from the database, to see if we can proceed with the update/delete.

Reply:
Got it, that sounds good - I don't think you need this field on the JSON body then, right?

Reply:
Yeah I think we can remove the field on JSON.

Also how do we want to enforce that an admin cannot have Edit or Delete book permission without Create book permission? I was thinking of just doing something like 
editBook <- createBook && editBook
Do we want to report an error instead?

Reply:
Yeah, I think it would be better to throw an error if the permissions get out of sync.